nodejs/eks/nodegroup.ts

```diff
@@ -755,35 +755,40 @@ ${customUserData}
         nodeAssociatePublicIpAddress = args.nodeAssociatePublicIpAddress;
     }
 
-    const numeric = new RegExp("^d+$");
+    const numeric = new RegExp("^\\d+$");
```

Comment:
Could use a separate PR as slightly unrelated, but LGTM.

Reply:
Agreed. I've included it in this PR since it's a small change and don't want to go through a long CI process. I plan to rebase and merge so the commits can be rolled back easily should we need to.